### PR TITLE
[wni] [aws] Open SSH port for Windows instance created

### PR DESCRIPTION
**Pull request by ravisantoshgudimetla**
This commit opens SSH port for Windows instance created
This ensures that we can transfer binaries over SCP
to the Windows instance, so that we can run our CI
tests.

Tested manually on my cluster. Need to verify if the e2es pass

/cc @aravindhp @sebsoto @vinaykns

### CODE REVIEW

File: tools/windows-node-installer/pkg/cloudprovider/aws/ec2.go [hunk 1]
```diff
@@ -146,6 +150,7 @@ func (a *AwsProvider) CreateWindowsVM() (credentials *types.Credentials, rerr er
         $file = "$env:temp\ConfigureRemotingForAnsible.ps1"
         (New-Object -TypeName System.Net.WebClient).DownloadFile($url,  $file)
         powershell.exe -ExecutionPolicy ByPass -File $file
+		powershell -NonInteractive -ExecutionPolicy Bypass Add-WindowsCapability -Online -Name OpenSSH.Server~~~~0.0.1.0
```
Comment (ravisantoshgudimetla) [inline tools/windows-node-installer/pkg/cloudprovider/aws/ec2.go]: For some reason, this is looking properly aligned locally for me.

File: tools/windows-node-installer/pkg/cloudprovider/aws/ec2.go [hunk 2]
```diff
@@ -37,6 +37,10 @@ const (
 	//  after launching an instance before trying to retrieve the generated password.`
 	// Ref: https://godoc.org/github.com/aws/aws-sdk-go/service/ec2#EC2.GetPasswordData
 	awsPasswordDataTimeOut = 15 * time.Minute
+	// sshPort to access the OpenSSH server installed on the windows node. This is needed
+	// for our CI testing.
+	// TODO: Make this optional and default opt out when we distribute the tool to customers
```
Comment (aravindhp) [inline tools/windows-node-installer/pkg/cloudprovider/aws/ec2.go]: We don't plan to distribute this tool to customers. Please remove this TODO.

File: tools/windows-node-installer/pkg/cloudprovider/aws/ec2.go (hunk 1 above)
Comment (aravindhp) [reply tools/windows-node-installer/pkg/cloudprovider/aws/ec2.go]: Enable whitspace display on your IDE and that might show you why.

Please update the comment for `userDataWinrm` variable.

File: tools/windows-node-installer/pkg/cloudprovider/aws/ec2.go [hunk 3]
```diff
@@ -435,13 +440,13 @@ func (a *AwsProvider) findOrCreateSg(infraID string, vpc *ec2.Vpc) (string, erro
 	log.Info(fmt.Sprintf("Using existing Security Group: %s.", sgID))
 
 	// Check winrm port open status for the exist security group
```
Comment (aravindhp) [inline tools/windows-node-installer/pkg/cloudprovider/aws/ec2.go]: s/exist/existing given you are changing the variable name.

File: tools/windows-node-installer/pkg/cloudprovider/aws/ec2.go [hunk 4]
```diff
@@ -451,9 +456,48 @@ func (a *AwsProvider) findOrCreateSg(infraID string, vpc *ec2.Vpc) (string, erro
 		log.Info(fmt.Sprintf("Winrm port already opened for Security Group: %s.", sgID))
 	}
 
+	// Check ssh port is open
```
Comment (aravindhp) [inline tools/windows-node-installer/pkg/cloudprovider/aws/ec2.go]: `Check if ssh port is open`

File: tools/windows-node-installer/pkg/cloudprovider/aws/ec2.go [hunk 5]
```diff
@@ -451,9 +456,48 @@ func (a *AwsProvider) findOrCreateSg(infraID string, vpc *ec2.Vpc) (string, erro
...
+		log.Info(fmt.Sprintf("ssh port is now added to Security Group %s", sgID))
+	} else {
+		log.Info(fmt.Sprintf("ssh port already opened for Security Group: %s.", sgID))
+	}
+
 	return sgID, nil
 }
 
+// checkSgSSHPortIsOpen checks whether the ssh port is opened in the given security group.
+func (a *AwsProvider) checkSgSSHPortIsOpen(sgId string) (bool, error) {
```
Comment (aravindhp) [inline tools/windows-node-installer/pkg/cloudprovider/aws/ec2.go]: Please rename to `checkSSHPortIsOpenInSG` and a TODO to make this a common helper for checking if ports are open in SGs.

File: tools/windows-node-installer/pkg/cloudprovider/aws/ec2.go [hunk 6]
```diff
@@ -506,12 +550,41 @@ func (a *AwsProvider) addWinrmPortToSg(sgId string) error {
 	return nil
 }
 
+// addSSHPortToSG creates inbound rule to allow traffic on ssh port for the given security group. If the port
+// doesn't not exist in the security group notify the user and generate one for the user.
+func (a *AwsProvider) addSSHPortToSG(sgId string) error {
```
Comment (aravindhp) [inline tools/windows-node-installer/pkg/cloudprovider/aws/ec2.go]: Add a TODO somewhere to modularize this

File: tools/windows-node-installer/test/e2e/aws_test.go [hunk 7]
```diff
@@ -19,7 +19,6 @@ var (
 	awscredentials = os.Getenv("AWS_SHARED_CREDENTIALS_FILE")
 	dir            = os.Getenv("ARTIFACT_DIR")
 	privateKeyPath = os.Getenv("KUBE_SSH_KEY_PATH")
-
```
Comment (aravindhp) [inline tools/windows-node-installer/test/e2e/aws_test.go]: Please revert

Comment (ravisantoshgudimetla) [reply tools/windows-node-installer/test/e2e/aws_test.go]: Done

**Pull request merged**